list_dependencies/deps.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+import subprocess
+import json
+
+def format_list(mylist):
+    if len(mylist) < 2:
+        return str(mylist)[1:-1]
+    else:
+        return str(mylist[:-1])[1:-1] + " and '" + str(mylist[-1]) + "'"
+
+def get_node_dependencies():
+    node_str = subprocess.run(
+            ['tilt', 'dump', 'engine'], 
+            stdout=subprocess.PIPE,
+        ).stdout.decode('utf-8')
+    node_data = json.loads(node_str)["ManifestTargets"]
+
+    dependencies = {}
+
+    for resInfo in node_data.values():
+        manifest = resInfo["Manifest"]
+        resName = manifest["Name"]
+        resDeps = manifest["ResourceDependencies"]
+        if resDeps == None:
+            resDeps = []
+        dependencies[resName] = resDeps
+
+    return dependencies
+    
+def get_node_status():
+    ready_str = subprocess.run(
+            ['tilt', 'get', 'kubernetesdiscovery', '-o', 'json'],
+            stdout=subprocess.PIPE,
+        ).stdout.decode('utf-8')
+    ready_data = json.loads(ready_str)["items"]
+
+    ready = {}
+
+    for resInfo in ready_data:
+        resName = resInfo["metadata"]["name"]
+        pods = resInfo["status"]["pods"]
+        resStatus = False
+        if pods != None:
+            for pod in pods:
+                containers = pod["containers"]
+                for container in containers:
+                    resStatus = container["ready"]
```

Comment:
@nicks is it possible to have multiple pods for a resource? If so, do all pods for a resource need to be ready before dependents are allowed to build? Maybe I should change this line to `resStatus &= container["ready"]` to make sure the status reflects the entire resource.

Reply:
for this demo, it's ok to assume that if one pod is ready, then anything that depends on it can start building.

the real relationship between readiness and resource dependencies is a bit slippery. pods can go from crashing to ready back to crashing. whereas once the resource starts building, we don't try to claw it back if the stuff it depends on starts crashing.

Reply:
I'll add a note about this in the README. Thank you!